doc/Development_Documentation/02_MVC/02_Template/02_Template_Extensions/README.md

```diff
@@ -0,0 +1,288 @@
+| `query`      |  array       | Optional params added to the request query string |
+
+   
+##### Example
+
+```twig
+<section id="foo-bar">
+    {{ pimcore_action('foo', 'Bar', null, { awesome: 'value' }) }}
+</section>
+```
+   
+    
+### `pimcore_cache`
+This is an implementation of an in-template cache. You can use this to cache some parts directly in the template, 
+independent of the other global definable caching functionality. This can be useful for templates which need a lot 
+of calculation or require a huge amount of objects (like navigations, ...).
+
+`pimcore_cache( name, lifetime, force)`
+
+| Name         | Type         | Description  |
+|--------------|--------------|--------------|
+| `name`       | string       | Name of cache item |
+| `lifetime`   | int          | Lifetime in seconds. If you define no lifetime the behavior is like the output cache, so if you make any change in Pimcore, the cache will be flushed. When specifying a lifetime this is independent from changes in the CMS. |
+| `force`      | bool         | Force caching, even when request is done within Pimcore admin interface |
+
+##### Example
+
+```twig
+{% set cache = pimcore_cache("test_cache_key", 60) %}
+{% if not cache.start() %}
+    <h1>This is some cached microtime</h1>
+    {{ 'now'|date('U') }}
+    {% do cache.end() %}
+{% endif %}
+```
+
+### `pimcore_device`
+
+This extension makes it easy to implement "Adaptive Design" in Pimcore. 
+
+```twig
+{{ pimcore_device('a default value') }}
+```
+
+| Name         | Type         | Description  |
+|--------------|--------------|--------------|
+| `default`    | string       | Default if no device can be detected |
+
+##### Example
+```twig
+{% if pimcore_device().isPhone() %}
+    This is my phone content
+{% elseif pimcore_device().isTablet() %}
+    This text is shown on a tablet
+{% elseif pimcore_device().isDesktop() %}
+    This is for default desktop Browser
+{% endif %}
+```
+   
+For details also see [Adaptive Design](../../../19_Development_Tools_and_Details/21_Adaptive_Design_Helper.md).
+
+### `pimcoreglossary`
+
+The `pimcoreglossary` block replaces glossary terms. See [Glossary](../../../18_Tools_and_Features/21_Glossary.md) for details.
+
+```twig
+{% pimcoreglossary %}
+My content
+{% endpimcoreglossary %}
+``` 
+
+
+### `pimcore_inc` 
+Use `pimcore_inc()` to include documents (eg. snippets) within views. 
+This is especially useful for footers, headers, navigations, sidebars, teasers, ...
+
+`pimcore_inc(document, params, cacheEnabled)`
+
+| Name           | Type         | Description  |
+|----------------|--------------|--------------|
+| `document`     | PageSnippet &#124; int &#124; string  | Document to include, can be either an ID, a path or even the Document object itself |
+| `params`       |  array       | Is optional and should be an array with key value pairs like in `$this->action()` from ZF. |
```

Comment:
`...from ZF.` there is no Zend anymore ;)

Reply:
kicked it

thanks for your review 🙏